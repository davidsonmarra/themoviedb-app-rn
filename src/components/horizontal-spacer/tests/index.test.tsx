import React from 'react';
import HorizontalSpacer from '..';
import {render} from '@testing-library/react-native';

let mockWidth = 10;

const createComponent = () => render(<HorizontalSpacer width={mockWidth} />);

describe('HorizontalSpacer', () => {
  beforeEach(() => {
    mockWidth = 10;
  });

  it('renders correctly', () => {
    const tree = createComponent().toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with mockWidth correctly', () => {
    const instance = createComponent().root;
    expect(instance.props.style).toStrictEqual({width: mockWidth});
  });
});
