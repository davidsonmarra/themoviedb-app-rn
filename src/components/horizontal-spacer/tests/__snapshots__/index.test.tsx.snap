// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalSpacer renders correctly 1`] = `
<View
  style={
    {
      "width": 10,
    }
  }
/>
`;
